industrial_ci/src/util.sh

```diff
@@ -407,7 +407,10 @@ function ici_make_temp_dir {
 function ici_relocate_target_path {
   local tmp_src
   ici_make_temp_dir tmp_src
-  cp -a "$TARGET_REPO_PATH" "$tmp_src/"
+  mkdir -p "$tmp_src/$(basename "$TARGET_REPO_PATH")"
+  [ -d "$BASEDIR" ] && echo Signature: 8a477f597d28d172789f06886806bc55 > "${BASEDIR}/CACHEDIR.TAG"
```

Comment:
I would like to move this to the script of the Gitlab CI config.

Reply:
Would this fix also not be useful for github actions (and others) if someone places the basedir and/or ccache in a directory under the checked out repo?  While this is less likely because those other systems allow you to cache directories outside the repo this issue would still happen on those ci systems if someone did.

To do this would you like me to make a different version of `ici_relocate_target_path` for gitlab or just add the part that writes the cache file in the gitlab ci script and leave the tar command here for the copy?

Reply:
> gitlab or just add the part that writes the cache file in the gitlab ci script and leave the tar command here for the copy?

Yes, I meant it this way

> Would this fix also not be useful for github actions (and others) if someone places the basedir and/or ccache in a directory under the checked out repo? 

It might be useful, but I am not sure, if this is an issue for most users.
Another idea would be to always mark these folder as caches, independent from `ici_relocate_target_path`.
We could use the `tar` command as well to speed up the copy to the source folder, even if `BASEDIR` is not set.